pkg/views/provider/list.go

```diff
@@ -9,176 +9,95 @@ import (
 )
 
-var columns = []table.Column{
-	{Title: "Name", Width: 20},
-	{Title: "Version", Width: 20},
+type RowData struct {
+	Name    string
+	Version string
 }
 
-type model struct {
-	table            table.Model
-	selectedProvider *serverapiclient.Provider
-	providers        map[string]serverapiclient.Provider
-	selectable       bool
-	initialRows      []table.Row
-}
-
-func (m model) Init() tea.Cmd {
-	if !m.selectable {
-		return tea.Quit
+func getRowFromRowData(rowData RowData) []string {
+	row := []string{
+		views.NameStyle.Render(rowData.Name),
+		views.DefaultRowDataStyle.Render(rowData.Version),
 	}
 
-	return nil
+	return row
 }
 
-func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
-	var cmd tea.Cmd
-	switch msg := msg.(type) {
-	case tea.WindowSizeMsg:
-		rows, cols := getRowsAndCols(msg.Width, m.initialRows)
-		m.table = getTable(rows, cols, m.selectable, m.table.Cursor())
-		return m, nil
-	case tea.KeyMsg:
-		switch msg.String() {
-		case "esc":
-			if m.table.Focused() {
-				m.table.Blur()
-			} else {
-				m.table.Focus()
-			}
-		case "q", "ctrl+c":
-			m.selectedProvider = nil
-			return m, tea.Quit
-		case "enter":
-			selectedProvider := m.providers[m.table.SelectedRow()[0]]
-			m.selectedProvider = &selectedProvider
-			return m, tea.Quit
-		}
-	}
-
-	m.table, cmd = m.table.Update(msg)
-	return m, cmd
-}
+func getRowData(provider *serverapiclient.Provider) RowData {
+	rowData := RowData{"", ""}
 
-var baseStyle = lipgloss.NewStyle().
-	BorderStyle(lipgloss.RoundedBorder()).
-	Padding(0, 1).MarginBottom(1)
+	rowData.Name = *provider.Name
+	rowData.Version = *provider.Version
 
-func (m model) View() string {
-	return baseStyle.Render(m.table.View()) + "\n"
+	return rowData
 }
 
-func renderProvidersList(providers []serverapiclient.Provider, selectable bool) model {
-	rows := []table.Row{}
-	selectedProvider := &providers[0]
-
-	for _, provider := range providers {
-		row := table.Row{*provider.Name, *provider.Version}
-		rows = append(rows, row)
-	}
-
-	width, _, _ := term.GetSize(int(os.Stdout.Fd()))
+func List(providerList []serverapiclient.Provider) {
 
-	adjustedRows, adjustedCols := getRowsAndCols(width, rows)
+	re := lipgloss.NewRenderer(os.Stdout)
 
-	providerMap := map[string]serverapiclient.Provider{}
-	for _, provider := range providers {
-		providerMap[*provider.Name] = provider
-	}
+	headers := []string{"Name", "Version"}
 
-	return model{
-		table:            getTable(adjustedRows, adjustedCols, selectable, 0),
-		selectedProvider: selectedProvider,
-		selectable:       selectable,
-		providers:        providerMap,
-		initialRows:      rows,
-	}
-}
+	data := [][]string{}
 
-func List(providers []serverapiclient.Provider) {
-	util.RenderMainTitle("PROVIDERS")
+	for _, provider := range providerList {
+		var rowData RowData
+		var row []string
 
-	if len(providers) == 0 {
-		view_util.RenderInfoMessage("No providers found")
-		return
+		rowData = getRowData(&provider)
+		if rowData == (RowData{}) {
```

Comment:
Can we avoid this kind of check? Why don't we use pointers here? (Applies to all other uses of `== RowData.{}` as well)

Reply:
no problem